ddtrace/appsec/_asm_request_context.py

```diff
@@ -225,23 +234,23 @@ def set_body_response(body_response):
     set_waf_address(SPAN_DATA_NAMES.RESPONSE_BODY, lambda: parse_response_body(body_response))
 
 
-def set_waf_address(address: str, value: Any, span: Optional[Span] = None) -> None:
+def set_waf_address(address: str, value: Any) -> None:
     if address == SPAN_DATA_NAMES.REQUEST_URI_RAW:
         parse_address = parse.urlparse(value)
         no_scheme = parse.ParseResult("", "", *parse_address[2:])
         waf_value = parse.urlunparse(no_scheme)
         set_value(_WAF_ADDRESSES, address, waf_value)
     else:
         set_value(_WAF_ADDRESSES, address, value)
-    if span is None:
-        span = _get_asm_context().span
-    if span:
-        core.set_item(address, value, span=span)
+    env = _get_asm_context()
+    if env and env.span:
+        root = env.span._local_root or env.span
+        root._set_ctx_item(address, value)
```

Comment:
Why'd you make the changes on these four lines?

Reply:
First span is no more a argument of the function and needs to be retrieved. Second set_item with span argument is deprecated. I did not change the logic (this is what set_item with span is doing), but anyway, I want to remove all set_ctx_item in a further change.